db/gdocBlockParsersRawToEnriched.ts

```diff
@@ -0,0 +1,642 @@
+        caption: Span[] = []
+    ): EnrichedBlockChart => ({
+        type: "chart",
+        url,
+        height,
+        row,
+        column,
+        position,
+        caption,
+        parseErrors: [error],
+    })
+
+    const val = chart.value
+
+    if (typeof val === "string") {
+        return {
+            type: "chart",
+            url: val,
+            height: undefined,
+            row: undefined,
+            column: undefined,
+            position: undefined,
+            caption: [],
+            parseErrors: [],
+        }
+    } else {
+        if (!val.url)
+            return createError(
+                {
+                    message: "url property is missing",
+                },
+                ""
+            )
+
+        const url = val.url
+
+        const warnings: ParseError[] = []
+
+        const height = val.height ? val.height : undefined
+        const row = val.row ? val.row : undefined
+        const column = val.column ? val.column : undefined
+        let position: ChartPositionChoice | undefined = undefined
+        if (val.position)
+            if (val.position === "featured") position = val.position
+            else {
+                warnings.push({
+                    message: "position must be 'featured' or unset",
+                })
+            }
+        const caption = val.caption ? htmlToSpans(val.caption) : []
+
+        return {
+            type: "chart",
+            url,
+            height,
+            row,
+            column,
+            position,
+            caption,
+            parseErrors: [],
+        }
+    }
+}
+
+const parseScroller = (scroller: RawBlockScroller): EnrichedBlockScroller => {
+    const createError = (
+        error: ParseError,
+        blocks: EnrichedScrollerItem[] = []
+    ): EnrichedBlockScroller => ({
+        type: "scroller",
+        blocks,
+        parseErrors: [error],
+    })
+
+    if (typeof scroller.value === "string")
+        return createError({
+            message: "Value is a string, not an object with properties",
+        })
+
+    const blocks: EnrichedScrollerItem[] = []
+    let currentBlock: EnrichedScrollerItem = {
+        url: "",
+        text: { type: "text", value: [], parseErrors: [] },
+    }
+    const warnings: ParseError[] = []
+    for (const block of scroller.value) {
+        match(block)
+            .with({ type: "url" }, (url) => {
+                if (currentBlock.url !== "") {
+                    blocks.push(currentBlock)
+                    currentBlock = {
+                        url: "",
+                        text: {
+                            type: "text",
+                            value: [],
+                            parseErrors: [],
+                        },
+                    }
+                }
+                currentBlock.url = url.value
+            })
+            .with({ type: "text" }, (text) => {
+                currentBlock.text = htmlToEnrichedTextBlock(text.value)
```

Comment:
As far as I understand ArchieML then this will be a list of items and several blocks of text are scooped up together with one image at a time. Url being set is the indicator that a new block is starting. Am I missing something?